### PR TITLE
Add index of Shimcache entry to ShimcacheRecord

### LINKED ISSUE #63: Shimcache plugin entry_index
As stated in the [source](https://andreafortuna.org/2017/10/16/amcache-and-shimcache-in-forensic-analysis/) of the [Shimcache](https://github.com/fox-it/dissect.target/blob/main/dissect/target/plugins/os/windows/regf/shimcache.py#L318) plugin:

> The events in Shimcache.hve are listed in chronological order with the most
recent event first and can be used in timelines to recreate and determine malicious activities.

Add the field `index` to the record `ShimcacheRecord` that represents the chronological order from `0` (the last shimmed entry) and upwards.

The field `index` can indicate which paths have been shimmed in chronological order.

### PR DESCRIPTION
**Pull request by Zawadidone**
Fixes #63

### CODE REVIEW

File: dissect/target/plugins/os/windows/regf/shimcache.py [hunk 1]
```diff
@@ -15,6 +15,8 @@
 ShimcacheRecord = TargetRecordDescriptor(
     "windows/shimcache",
     [
+        ("string", "name"),
+        ("varint", "entry_index"),
```
Comment (Schamper) [inline dissect/target/plugins/os/windows/regf/shimcache.py:19]:
```suggestion
        ("varint", "index"),
```
I think "entry" is already implied and thus redundant.

File: dissect/target/plugins/os/windows/regf/shimcache.py [hunk 2]
```diff
@@ -320,6 +322,8 @@ def shimcache(self) -> ShimcacheRecord:
         Yields ShimcacheRecords with the following fields:
             hostname (string): The target hostname.
             domain (string): The target domain.
+            name (string): The value name.
+            entry_index (varint): The index of the entry.
```
Comment (Schamper) [inline dissect/target/plugins/os/windows/regf/shimcache.py:326]:
```suggestion
            index (varint): The index of the entry.
```

File: dissect/target/plugins/os/windows/regf/shimcache.py [hunk 3]
```diff
@@ -339,12 +343,12 @@ def shimcache(self) -> ShimcacheRecord:
                     self.target.log.warning("Error parsing ShimCache entry: %s %s", key, value_name)
                     continue
 
-                yield from self._get_records(cache)
+                yield from self._get_records(cache, value_name)
```
Comment (Schamper) [inline dissect/target/plugins/os/windows/regf/shimcache.py:346]:
```suggestion
                yield from self._get_records(value_name, cache)
```
Nitpicking, but you're passing the cache opened from `value_name`, so putting `value_name` first feels more natural.

File: dissect/target/plugins/os/windows/regf/shimcache.py [hunk 4]
```diff
@@ -339,12 +343,12 @@ def shimcache(self) -> ShimcacheRecord:
...
 
     def _get_records(
-        self, cache: Generator[ShimCacheGeneratorType, None, None]
+        self, cache, name: Generator[ShimCacheGeneratorType, None, None]
```
Comment (Schamper) [inline dissect/target/plugins/os/windows/regf/shimcache.py:349]:
```suggestion
        self, name: str, cache: Generator[ShimCacheGeneratorType, None, None]
```
Moving the order + fixing the type hint.

File: dissect/target/plugins/os/windows/regf/shimcache.py [hunk 5]
```diff
@@ -354,6 +358,8 @@ def _get_records(
             path = uri.from_windows(self.target.resolve(path))
 
             yield ShimcacheRecord(
+                name=name,
+                entry_index=index+1,
```
Comment (Schamper) [inline dissect/target/plugins/os/windows/regf/shimcache.py:362]: Why not start counting at 0?

**Review by Schamper — CHANGES_REQUESTED** [review]

**Comment by Zawadidone** [comment]
I will adjust the tests later on.

**Review by Schamper — APPROVED** [review]

**Pull request merged**